framework/moose.mk

```
@@ -38,17 +38,25 @@ pcre_deps      := $(patsubst %.cc, %.$(obj-suffix).d, $(pcre_srcfiles)) \
 #
 # hit (new getpot parser)
 #
-hit_DIR       := $(FRAMEWORK_DIR)/contrib/hit
-hit_srcfiles  := $(hit_DIR)/parse.cc $(hit_DIR)/lex.cc $(hit_DIR)/braceexpr.cc
+HIT_DIR ?= $(MOOSE_DIR)/moosetools/contrib/hit
+ifeq ($(wildcard $(HIT_DIR)), "")
+        HIT_DIR = $(MOOSE_DIR)/framework/contrib/hit
+endif
+$(info Using HIT from $(HIT_DIR))
+hit_CONTENT   := $(shell ls $(HIT_DIR) 2> /dev/null)
+ifeq ($(hit_CONTENT),)
+  $(error The HIT input file parser does not seem to be available. Make sure that either the moosetools submodule is checked out or that your HIT_DIR environment variable is set to the correct location of your HIT parser. To get the moosetools submodule, run the following from the root directory of MOOSE.\n    scripts/update_and_rebuild_hit.sh)
```

```suggestion
  $(error The HIT input file parser does not seem to be available. Make sure that either the moosetools submodule is checked out or that your HIT_DIR environment variable is set to the correct location of your HIT parser. To get the moosetools submodule, run the following from the root directory of MOOSE.\n\n    scripts/update_and_rebuild_hit.sh)
```

We should never hit this for now, right?

---

Correct. This would be if you set HIT_DIR to something that does not exist.